wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/Components.java

```diff
@@ -51,7 +51,12 @@ public static String getIdentifiableComponentName(Component component)
         return component.getImage();
       case KUBERNETES_COMPONENT_TYPE:
       case OPENSHIFT_COMPONENT_TYPE:
-        return component.getReference();
+        String reference = component.getReference();
+        if (reference != null) {
```

Comment:
Oh. why not
`return reference != null ? reference : component.getType()`

Reply:
:smile_cat: